### PR TITLE
chore(hack): upgrade Trivy to v0.56.1 for consistency with CI

### PR DESCRIPTION
**Pull request by MAVRICK-1**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
- Updated Trivy Version: Changed the version from v0.48.1 to v0.53.0 to ensure consistency across local usage and GitHub CI.

**Which issue(s) this PR fixes**:
Part of https://github.com/karmada-io/karmada/issues/5484 update [hack/scan-image-vuln.sh](https://github.com/karmada-io/karmada/blob/master/hack/scan-image-vuln.sh)

**Special notes for your reviewer**:
- This change aligns the Trivy version used in the script with the version used in the CI workflow.

**Does this PR introduce a user-facing change?**

```release-note
NONE

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/assign @zhzhuang-zju

File: hack/scan-image-vuln.sh [hunk 1]
```diff
@@ -78,7 +78,7 @@ if util::cmd_exist trivy ; then
 	echo "pass"
 else
 	echo "start installing trivy"
-	curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.48.1
+	curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.53.0
```
Comment (zhzhuang-zju) [inline hack/scan-image-vuln.sh:81]:
```suggestion
	curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.56.1
```
`trivy-action` has supported `trivy v0.56.1` starting from version `v0.25.0`. Please update it, and then LGTM~

Comment (MAVRICK-1) [reply hack/scan-image-vuln.sh]: @zhzhuang-zju rectified

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
@MAVRICK-1 would you mind [squashing the commits](https://karmada.io/docs/contributor/github-workflow/#squash-commits)?

**Comment by MAVRICK-1** [comment]
Okay 👍

**Comment by zhzhuang-zju** [comment]
/retitle chore(hack): upgrade Trivy to v0.56.1 for consistency with CI

**Comment by MAVRICK-1** [comment]
@RainbowMango done

**Comment by zhzhuang-zju** [comment]
thanks
/hold cancel

**Review by RainbowMango — COMMENTED** [review]
/lgtm

**Pull request merged**